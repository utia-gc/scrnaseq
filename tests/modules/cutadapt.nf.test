nextflow_process {

    name "Test Process cutadapt"
    script "modules/cutadapt.nf"
    process "cutadapt"

    test("cutadapt works for SE reads.") {

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = [
                    [
                        sampleName: "SRR1066657_GSM1299413_WT_NR_A",
                        readType:   "single"
                    ],
                    [
                        file("https://github.com/utia-gc/ngs-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz")
                    ]
                ]
                input[1] = "AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT"
                input[2] = "AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC"
                input[3] = "30"
                """
            }
        }

        then {
            assert process.success

            with(process.out.reads) {
                assert size() == 1
                assert get(0).get(1) ==~ /^.*\/SRR1066657_GSM1299413_WT_NR_A_trimmed_R1.fastq.gz/
            }
            with(process.out.log) {
                assert size() == 1
                assert get(0) ==~ /^.*\/SRR1066657_GSM1299413_WT_NR_A_cutadapt-log.txt/
            }

            assert snapshot(process.out.reads).match()
        }

    }

    test("cutadapt works for PE reads.") {

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = [
                    [
                        sampleName: "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                        readType:   "paired"
                    ],
                    [
                        file("https://github.com/utia-gc/ngs-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz"),
                        file("https://github.com/utia-gc/ngs-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz")
                    ]
                ]
                input[1] = "AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT"
                input[2] = "AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC"
                input[3] = "30"
                """
            }
        }

        then {
            assert process.success

            with(process.out.reads) {
                assert size() == 1
                assert get(0).get(1).get(0) ==~ /^.*\/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_trimmed_R1.fastq.gz/
                assert get(0).get(1).get(1) ==~ /^.*\/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_trimmed_R2.fastq.gz/
            }
            with(process.out.log) {
                assert size() == 1
                assert get(0) ==~ /^.*\/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_cutadapt-log.txt/
            }

            assert snapshot(process.out.reads).match()
        }
    }

}
