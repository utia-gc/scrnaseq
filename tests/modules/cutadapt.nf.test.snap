{
    "cutadapt works for single-end reads and publishes trimmed reads when `params.publishTrimReads = true`.": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                        "readType": "single"
                    },
                    "SRR1066657_GSM1299413_WT_NR_A_trimmed_R1.fastq.gz:md5,9c46e73133a169e413c6f94eb6afb277"
                ]
            ]
        ],
        "timestamp": "2023-08-17T20:03:28+0000"
    },
    "CUTADAPT works for single-end reads.": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                        "readType": "single"
                    },
                    "SRR1066657_GSM1299413_WT_NR_A_R1.fastq.gz:md5,9c46e73133a169e413c6f94eb6afb277"
                ]
            ]
        ],
        "timestamp": "2023-08-17T20:03:28+0000"
    },
    "CUTADAPT works for paired-end reads.": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                        "readType": "paired"
                    },
                    [
                        "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_R1.fastq.gz:md5,ce49af7bca8aedfc501c2b805034c7f4",
                        "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_R2.fastq.gz:md5,c8ae59bdc21d387338e752c1ed066bb2"
                    ]
                ]
            ]
        ],
        "timestamp": "2023-08-17T20:03:28+0000"
    },
    "cutadapt works for paired-end reads.": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                        "readType": "paired"
                    },
                    [
                        "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_R1.fastq.gz:md5,ce49af7bca8aedfc501c2b805034c7f4",
                        "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_R2.fastq.gz:md5,c8ae59bdc21d387338e752c1ed066bb2"
                    ]
                ]
            ]
        ],
        "timestamp": "2023-08-17T20:03:28+0000"
    },
    "cutadapt works for single-end reads.": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                        "readType": "single"
                    },
                    "SRR1066657_GSM1299413_WT_NR_A_R1.fastq.gz:md5,9c46e73133a169e413c6f94eb6afb277"
                ]
            ]
        ],
        "timestamp": "2023-08-17T20:03:28+0000"
    },
    "cutadapt works for SE reads.": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                        "readType": "single"
                    },
                    "SRR1066657_GSM1299413_WT_NR_A_trimmed_R1.fastq.gz:md5,9c46e73133a169e413c6f94eb6afb277"
                ]
            ]
        ],
        "timestamp": "2023-08-17T20:03:28+0000"
    },
    "cutadapt works for PE reads.": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                        "readType": "paired"
                    },
                    [
                        "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_trimmed_R1.fastq.gz:md5,ce49af7bca8aedfc501c2b805034c7f4",
                        "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_trimmed_R2.fastq.gz:md5,c8ae59bdc21d387338e752c1ed066bb2"
                    ]
                ]
            ]
        ],
        "timestamp": "2023-08-17T20:03:28+0000"
    }
}